### ISSUE TITLE
Let USRP interface retry a few times if it doesn't get requested number of samples

### ISSUE THREAD

**Issue by djanderson** [NTIA/scos-sensor#125]
The USRP occasionally hiccups and doesn't return the requested number of samples. @ToddSchumann 's tests indicate that just re-requesting the samples succeeds in almost all cases. The SCOS USRP interface should define MAX_RETRIES (a low number, 5 or so should probably be fine), and retry the acquisition that many times before throwing the RuntimeError. 

Make sure that the database query for `scale_factor` happens only after the samples have been successfully acquired.

https://github.com/NTIA/scos-sensor/blob/master/src/actions/usrp.py#L153

**Comment by ToddSchumann** [comment]
I ran a test to see how these errors occurred. They seem to come with overflows/underflows, indicating that a USB frame is dropped as the bus can't keep up with the incoming stream. To characterize the process and its recoverability, I ran the following test:

Clock/sampling frequency set to 48MHz to run the bus as fast as possible (the sample rate can go up to 56MHz, but 48MHz was as fast as I've characterized, so I limited it here). Repeatedly take data acquisitions until 5 errors (not necessarily sequential) occur. The acquisition was for 1024 samples after 10,000,000 conditioning samples (at each acquisition). The resultant output is:
`Reading 1024 samples after 10000000 conditioning samples...
OEncountered error:
Data length 0/1024
Retrying 5 more times...
Reached 100 data acquisitions with 1 errors
Reached 200 data acquisitions with 1 errors
OEncountered error:
Data length 0/1024
Retrying 4 more times...
Reached 300 data acquisitions with 2 errors
Reached 400 data acquisitions with 2 errors
Reached 500 data acquisitions with 2 errors
Reached 600 data acquisitions with 2 errors
Reached 700 data acquisitions with 2 errors
OEncountered error:
Data length 0/1024
Retrying 3 more times...
Reached 800 data acquisitions with 3 errors
Reached 900 data acquisitions with 3 errors
OEncountered error:
Data length 0/1024
Retrying 2 more times...
Reached 1000 data acquisitions with 4 errors
Reached 1100 data acquisitions with 4 errors
Reached 1200 data acquisitions with 4 errors
OEncountered error:
Data length 0/1024
Retrying 1 more times...
Reached 1300 data acquisitions with 5 errors
Reached 1400 data acquisitions with 5 errors
Reached 1500 data acquisitions with 5 errors
Reached 1600 data acquisitions with 5 errors
Reached 1700 data acquisitions with 5 errors
Reached 1800 data acquisitions with 5 errors
Reached 1900 data acquisitions with 5 errors
Reached 2000 data acquisitions with 5 errors
OEncountered error:
Data length 0/1024
Failed too many times...
Took 2047 measurements before erroring 5 times...`

In the test, no two errors occured sequentially, so a low number for retries should be sufficient. If the retries fail further, it is likely that something more serious is occurring. I'd suggest attempting to reconnect in this case.

Note: The 'O' prior to the "Encountered error:" in the example output indicated that the USB frame encountered an overflow. In these cases, the USB frame appears to be dropped as the data length is exactly 0 instead of the expected 1024. The likelihood of this happening should increase with sampling rate.
